llvm/lib/Analysis/UniformityAnalysis.cpp

```diff
@@ -32,7 +32,10 @@ bool llvm::GenericUniformityAnalysisImpl<SSAContext>::markDefsDivergent(
 
 template <> void llvm::GenericUniformityAnalysisImpl<SSAContext>::initialize() {
   for (auto &I : instructions(F)) {
-    if (TTI->isSourceOfDivergence(&I))
+    bool NoDivergenceSource = false;
+    if (auto Call = dyn_cast<CallBase>(&I))
+      NoDivergenceSource = Call->hasFnAttr(Attribute::NoDivergenceSource);
```

Comment:
Would it make more sense to check for calls in `TargetTransformInfo::isSourceOfDivergence` before it calls into `TTIImpl`?

Reply:
Yes. :-)